File: sla/sla.java
import java.util.Scanner;

public class Calculadora {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        double num1, num2, resultado;
        char op;

        System.out.print("Digite o primeiro numero: ");
        num1 = sc.nextDouble();

        System.out.print("Digite o operador (+, -, *, /): ");
        op = sc.next().charAt(0); // pega o primeiro caractere digitado

        System.out.print("Digite o segundo numero: ");
        num2 = sc.nextDouble();

        if (op == '+') {
            resultado = num1 + num2;
        } else if (op == '-') {
            resultado = num1 - num2;
        } else if (op == '*') {
            resultado = num1 * num2;
        } else if (op == '/') {
            if (num2 != 0) {
                resultado = num1 / num2;
            } else {
                System.out.println("Erro: divisao por zero!");
                sc.close();
                return;
            }
        } else {
            System.out.println("Operador invalido!");
            sc.close();
            return;
        }

        System.out.println("Resultado: " + resultado);
        sc.close();
    }
}

File: sla/sla.f90
program calculadora
  implicit none
  real :: num1, num2, resultado
  character :: op

  print *, "Digite o primeiro numero:"
  read *, num1
  print *, "Digite o segundo numero:"
  read *, num2
  print *, "Escolha a operacao (+ - * /):"
  read *, op

  select case (op)
  case ('+')
     resultado = num1 + num2
  case ('-')
     resultado = num1 - num2
  case ('*')
     resultado = num1 * num2
  case ('/')
     resultado = num1 / num2
  case default
     print *, "Operacao invalida."
     stop
  end select

  print *, "Resultado:", resultado
end program calculadora
